src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -1126,10 +1170,6 @@ private static ImmutableArray<StateForCase> RemoveEvaluation(ImmutableArray<Stat
             trueTestImpliesTrueOther = false;
             falseTestImpliesTrueOther = false;
 
-            // if the tests are for unrelated things, there is no implication from one to the other
-            if (!test.Input.Equals(other.Input))
```

Comment:
Should we put this back as an assertion? (if we get here, then the inputs are related)

Reply:
Done.